workspaces/adventure-pack/goodies/java/src/traverse_post_order/AP.java

```diff
@@ -0,0 +1,68 @@
+package traverse_post_order;
+
+import iterable_stream.IterableStream;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+import java.util.Stack;
+import pair.Pair;
+import simple_iterator.SimpleIterator;
+
+/**
+ * Post-order binary tree traversal implementation.
+ */
+public final class AP {
+
+  /**
+   * Get {@link IterableStream iterable stream} of all the {@link TreeNode nodes} in the binary tree
+   * in post-order.
+   * @param root Binary tree root {@link TreeNode node}.
+   * @return The {@link IterableStream iterable stream} in post-order.
+   */
+  public static IterableStream<TreeNode> traversePostOrder(
+    final TreeNode root
+  ) {
+    final Stack<Pair<TreeNode, Boolean>> stack = new Stack<
+      Pair<TreeNode, Boolean>
+    >();
+    stack.push(new Pair<>(root, false));
+    return IterableStream.from(
+      SimpleIterator.toIterator(() -> {
+        while (!stack.isEmpty()) {
+          Pair<TreeNode, Boolean> peek = stack.pop();
+          TreeNode<T> node = peek.first();
```

Comment:
Does this actually work as a generic? I see it defined below as containing ints.

Reply:
Ah.. sorry, it does not. my bad. I'll fix.

Reply:
Resolved.